diff --git a/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt b/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
index 6f45884..ae7fbb5 100644
--- a/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
+++ b/node_modules/react-native-svg/android/src/main/jni/CMakeLists.txt
@@ -31,6 +31,7 @@ find_package(ReactAndroid REQUIRED CONFIG)
 if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
   target_link_libraries(
     react_codegen_rnsvg
+    c++_shared
     ReactAndroid::reactnative
     ReactAndroid::jsi
   )
