diff --git a/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt b/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt
index 026e7ca..651cc50 100644
--- a/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt
+++ b/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt
@@ -35,6 +35,7 @@ find_package(fbjni REQUIRED CONFIG)
 
 target_link_libraries(
   ${PACKAGE_NAME}
+  c++_shared
   ReactAndroid::reactnative
   ReactAndroid::jsi
   fbjni::fbjni
