diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java
index 82d815c..7e83179 100644
--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java
+++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java
@@ -188,7 +188,9 @@ public class ReactActivityDelegate {
   }
 
   public void onUserLeaveHint() {
-    Objects.requireNonNull(mReactDelegate).onUserLeaveHint();
+    if (mReactDelegate != null) {
+      mReactDelegate.onUserLeaveHint();
+    }
   }
 
   public void onPause() {
